runtime/doc/lua.txt

```diff
@@ -2144,4 +2144,102 @@ set({mode}, {lhs}, {rhs}, {opts})                           *vim.keymap.set()*
                 See also: ~
                     |nvim_set_keymap()|
 
+
+==============================================================================
+Lua module: path                                                    *lua-path*
+
+basename({file})                                         *vim.path.basename()*
+                Return the basename of the given file or directory
+
+                Parameters: ~
+                    {file}  (string) File or directory
+
+                Return: ~
+                    (string) Basename of {file}
+
+dir({path})                                                   *vim.path.dir()*
+                Return an iterator over the files and directories located in
+                {path}
+
+                Parameters: ~
+                    {path}  (string) Directory
+
+                Return: ~
+                    Iterator over files and directories in {path}. Each
+                    iteration yields two values: name and type.
+
+dirname({file})                                           *vim.path.dirname()*
+                Return the parent directory of the given file or directory
+
+                Parameters: ~
+                    {file}  (string) File or directory
+
+                Return: ~
+                    (string) Parent directory of {file}
+
+find({names}, {kwargs})                                      *vim.path.find()*
```

Comment:
nit: should stick to `opt` until/unless we decide to use "kwargs" everywhere. (Though "kwargs" is a useful way to describe the idea in comments/discussions.)